hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2794,7 +2794,11 @@ public HoodieWriteConfig build() {
     private String getDefaultMarkersType(EngineType engineType) {
       switch (engineType) {
         case SPARK:
-          return MarkerType.TIMELINE_SERVER_BASED.toString();
+          if (writeConfig.isEmbeddedTimelineServerEnabled()) {
+            return MarkerType.TIMELINE_SERVER_BASED.toString();
+          } else {
+            return MarkerType.DIRECT.toString();
```

Comment:
we could log.warn here.

Reply:
done

Reply:
The logic of turning off the timeline-server-based markers is already covered by `WriteMarkersFactory`.